Add BookmarkList to Bookmarks app

pavan-testpress opened the pull request:
- Added BookmarksListView.
- Pagination used upto 10 records per page.
- TestCases added(test_bookmark_list).
- BookmarkList template added.
- Sort by name, modified, created added.
- Sorting order consistancy took care in pagination.

pavan-testpress committed:
Add FolderListView to BookmarksApp

- Add base template
- Add folderlist template
- Add css styling sheet
- Add index view
- Add FolderListView
- Add sort by name, created date, modified date

pavan-testpress committed:
Add FolderListView to BookmarksApp

- Add base template
- Add folderlist template
- Add css styling sheet
- Add index view
- Add FolderListView
- Add sort by name, created date, modified date

pavan-testpress committed:
Add TestCases for Authentication app, Bookmarksapp

- Add login, logout, signup testcases to aurhententication app
- Add testcase to access bookmarks without login
- Add testcase login
- Add testcase for FolderListView
- Add testcase for FolderLiseView sort by name
- Add testcase for FolderLiseView sort by created date
- Add testcase for FolderLiseView sort by modified date

pavan-testpress committed:
Add TestCases for Authentication app, Bookmarksapp

- Add login, logout, signup testcases to aurhententication app
- Add testcase to access bookmarks without login
- Add testcase login
- Add testcase for FolderListView
- Add testcase for FolderLiseView sort by name
- Add testcase for FolderLiseView sort by created date
- Add testcase for FolderLiseView sort by modified date

pavan-testpress committed:
Add TestCases for Authentication app, Bookmarksapp

- Add login, logout, signup testcases to aurhententication app
- Add testcase to access bookmarks without login
- Add testcase login
- Add testcase for FolderListView
- Add testcase for FolderLiseView sort by name
- Add testcase for FolderLiseView sort by created date
- Add testcase for FolderLiseView sort by modified date

pavan-testpress committed:
Verify Flake8 Conventions

pavan-testpress committed:
Verify Flake8 Conventions

pavan-testpress committed:
Verify Flake8 Conventions

pavan-testpress committed:
Add BookmarkList to Bookmarks app

- Add BookmarkList template
- Add BookmarkList View
- Add Sort by name, bookmark created & modifide feature
- Add Ubuntu fonts used in Css locally
- Add FontAwesome icons locally

pavan-testpress committed:
Add BookmarkList to Bookmarks app

- Add BookmarkList template
- Add BookmarkList View
- Add Sort by name, bookmark created & modifide feature
- Add Ubuntu fonts used in Css locally
- Add FontAwesome icons locally

pavan-testpress committed:
Add BookmarkList to Bookmarks app

- Add BookmarkList template
- Add BookmarkList View
- Add Sort by name, bookmark created & modifide feature
- Add Ubuntu fonts used in Css locally
- Add FontAwesome icons locally

pavan-testpress committed:
Add BookmarkList to Bookmarks app

- Add BookmarkList template
- Add BookmarkList View
- Add Sort by name, bookmark created & modifide feature
- Add Ubuntu fonts used in Css locally
- Add FontAwesome icons locally

pavan-testpress committed:
Add Bookmarkslist TestCases

pavan-testpress committed:
Add Bookmarkslist TestCases

pavan-testpress committed:
Remove Bookmarks TestCases

pavan-testpress committed:
Change Testcases naming convention to underscore

pavan-testpress committed:
Change Testcases naming convention to underscore

pavan-testpress committed:
Change Testcases naming convention to underscore

pavan-testpress committed:
Add FolderListView to BookmarksApp (#5)

* Add FolderListView to BookmarksApp

- Add base template
- Add folderlist template
- Add css styling sheet
- Add index view
- Add FolderListView
- Add sort by name, created date, modified date

* Add TestCases

* Bug Fix for Sorting in FolderListView

- Renamed testcases with underscore convention
- Fastened the hover effect to 0.3 sec
- Added sort by name as default if unknown sort order given

* Add comments to Bookmarks TestCases

pavan-testpress committed:
Add Comments to TestCases

pavan-testpress committed:
Add Comments to TestCases

pavan-testpress committed:
Merge branch 'testcases' of https://github.com/pavan-testpress/DjangoProject into testcases

pavan-testpress committed:
Add gitignore

pavan-testpress committed:
Add gitignore

pavan-testpress committed:
Fix git pushing

pavan-testpress committed:
Fix git pushing

pavan-testpress committed:
Merge branch 'bookmarkslist' of https://github.com/pavan-testpress/DjangoProject into bookmarkslist

pavan-testpress committed:
Add TestCases for Bookmarkslist & Fix Sorting Bug

- Added testcases for bookmarkslist
- handled invlaid sorting to go default by name

pavan-testpress committed:
Add TestCases for Bookmarkslist & Fix Sorting Bug

- Added testcases for bookmarkslist
- handled invlaid sorting to go default by name

waranlogesh commented on DjangoProject/bookmarksapp/tests.py:
```diff
@@ -6,69 +6,151 @@
+        """
+        Test folderlist page without logged in the user.
+        """
         self.client.get(reverse('authenticationapp:logout'))
```
Why logging out at the start of each test case? It should be taken care of by the teardown.

pavan-testpress replied:
For this testcase, I need to check which page it is rendering if user is not logged in. Since user is already logged in with setUp method and teardown will execute after completing this testcase. I am logging out inside this testcase.

waranlogesh commented on DjangoProject/bookmarksapp/urls.py:
```diff
@@ -7,4 +7,5 @@
 urlpatterns = [
     path('', views.index, name="index"),
     path('folders/', views.FolderListView.as_view(), name="folders"),
+    path('folders/<slug>/', views.BookmarksListView.as_view(), name="bookmarks")
```
It should be 
```suggestion
    path('folders/<slug>/', views.BookmarksListView.as_view(), name="folder-bookmarks")
```

pavan-testpress replied:
changed to folder-bookmarks

waranlogesh commented on DjangoProject/bookmarksapp/views.py:
```diff
@@ -17,12 +17,52 @@ class FolderListView(ListView):
     context_object_name = 'folders'
     template_name = 'bookmarksapp/folderlist.html'
 
+    def dispatch(self, request, *args, **kwargs):
```
Use `login_required` decorator.

pavan-testpress replied:
Used decorator

waranlogesh commented on DjangoProject/bookmarksapp/views.py:
```diff
@@ -17,12 +17,52 @@ class FolderListView(ListView):
+            sort = self.request.GET['sort']
+            if sort not in ['-modified', '-created', 'name']:
+                sort = 'name'
+                return sort
```
```suggestion
                return 'name'
```

pavan-testpress replied:
changed

waranlogesh commented on DjangoProject/bookmarksapp/views.py:
```diff
@@ -17,12 +17,52 @@ class FolderListView(ListView):
-            return self.request.GET['sort']
+            sort = self.request.GET['sort']
+            if sort not in ['-modified', '-created', 'name']:
+                sort = 'name'
```
```suggestion
```

pavan-testpress replied:
changed

waranlogesh commented on DjangoProject/bookmarksapp/views.py:
```diff
@@ -17,12 +17,52 @@ class FolderListView(ListView):
+                sort = 'name'
+                return sort
+            return sort
+        else:
```
```suggestion
```

waranlogesh commented on DjangoProject/bookmarksapp/views.py:
```diff
@@ -17,12 +17,52 @@ class FolderListView(ListView):
+                return sort
+            return sort
+        else:
+            return None
```
```suggestion
        return None
```

pavan-testpress replied:
changed

waranlogesh commented on DjangoProject/bookmarksapp/views.py:
```diff
@@ -17,12 +17,52 @@ class FolderListView(ListView):
+class BookmarksListView(ListView):
+    model = Folders
+    context_object_name = 'bookmarks'
+    template_name = 'bookmarksapp/bookmarkslist.html'
```
```suggestion
    template_name = 'bookmarksapp/bookmarks-list.html'
```

waranlogesh commented on DjangoProject/bookmarksapp/views.py:
```diff
@@ -17,12 +17,52 @@ class FolderListView(ListView):
+    context_object_name = 'bookmarks'
+    template_name = 'bookmarksapp/bookmarkslist.html'
+
+    def dispatch(self, request, *args, **kwargs):
```
Use `login-required` decorator.

pavan-testpress replied:
used decorator

waranlogesh commented on DjangoProject/bookmarksapp/views.py:
```diff
@@ -17,12 +17,52 @@ class FolderListView(ListView):
+            if sort not in ['-modified', '-created', 'name']:
+                sort = 'name'
+                return sort
+            return sort
```
Simplify according to a previous similar comment above.

waranlogesh commented on DjangoProject/bookmarksapp/views.py:
```diff
@@ -17,12 +17,52 @@ class FolderListView(ListView):
+
+    def get_queryset(self):
+        if 'sort' in self.request.GET:
+            self.queryset = Folders.objects.get(created_by=self.request.user, slug=self.kwargs['slug']).folder
```
Whats the purpose of this?

pavan-testpress replied:
i am using Folders model to get bookmarks related to it instead of bookmarks model.

pavan-testpress committed:
Use login required decorator(dispatch)

pavan-testpress committed:
Use login required decorator(dispatch)

pavan-testpress committed:
Add BookmarksList

- Added BookmarksListView.
- Pagination used upto 10 records per page.
- TestCases added(test_bookmark_list).
- BookmarkList template added.
- Sort by name, modified, created added.
- Sorting order consistancy took care in pagination.

pavan-testpress committed:
Merge branch 'bookmarkslist' of https://github.com/pavan-testpress/DjangoProject into bookmarkslist

pavan-testpress committed:
Deleted Authentication tests
